zerver/lib/events.py

```diff
@@ -1112,6 +1113,22 @@ def _draft_update_action(i: int) -> None:
                             for user_id in user_group["members"]
                             if user_id != person_user_id
                         ]
+
+                    for setting_name in Realm.REALM_PERMISSION_GROUP_SETTINGS_WITH_NEW_API_FORMAT:
+                        if not isinstance(state["realm_" + setting_name], int):
+                            state["realm_" + setting_name].direct_members = [
+                                user_id
+                                for user_id in state["realm_" + setting_name].direct_members
+                                if user_id != person_user_id
+                            ]
+                    for group in state["realm_user_groups"]:
+                        for setting_name in NamedUserGroup.GROUP_PERMISSION_SETTINGS:
+                            if not isinstance(group[setting_name], int):
+                                group[setting_name].direct_members = [
+                                    user_id
+                                    for user_id in group[setting_name].direct_members
+                                    if user_id != person_user_id
```

Comment:
Shouldn't there be something similar for Stream hanging settings?

Reply:
We currently do not have stream settings that support anonymous groups, so skipped that for now. But can add those as well.

Reply:
We will in the future, so it seems like we should just have the logic as one less thing to remember when we do those. Though maybe `test_events` will fail for us naturally.